### PR TITLE
feat: add anime rating annotator

### LINKED ISSUE #81: Evaluate new rating model candidates as model-native rating annotators
## Parent

Parent: #79

## 方針

この issue は既存実装済み WDTagger 系ではなく、今回調査した新規 rating model 候補を評価・必要なら追加するための sub issue。

image-annotator-lib 側は LoRAIro canonical rating (`PG/PG-13/R/X/XXX`) へ変換しない。モデルが出した raw label / confidence / source scheme を model-native rating として返すだけにする。

LoRAIro 側での canonical rating 変換・DB保存は NEXTAltair/LoRAIro#333 で扱う。

### 1. `deepghs/anime_rating`

- URL: https://huggingface.co/deepghs/anime_rating
- docs: https://dghs-imgutils.deepghs.org/main/api_doc/validate/rating.html
- labels: `safe`, `r15`, `r18`
- format: ONNX
- license: MIT
- model card 上の精度:
  - `caformer_s36_plus`: 74.26%
  - `mobilenetv3`: 64.77%
  - `mobilenetv3_sce`: 66.27%
  - `mobilenetv3_sce_dist`: 69.49%

評価観点:

- rating-only annotator として最有力
- ONNX なので lazy import 方針に合わせやすい
- `safe/r15/r18` の境界は曖昧。LoRAIro mapper は downstream 責務

### 2. `deepghs/eattach_sankaku_rating`

- URL: https://huggingface.co/deepghs/eattach_sankaku_rating
- labels: `r15`, `r18`, `safe`
- encoder: `SmilingWolf/wd-swinv2-tagger-v3`
- metrics:
  - AUC 0.8746
  - accuracy 0.7013
  - mAP 0.7853
- format: ONNX / Safetensors

評価観点:

- Sankaku 3分類系
- WD tagger 系 encoder 前提のため統合方式を確認する
- `deepghs/anime_rating` より導入が簡単か比較する

### 3. `Camais03/camie-tagger-v2`

- URL: https://huggingface.co/Camais03/camie-tagger-v2
- 種別: anime multi-label tagger
- rating category: 4 tags
- reported rating performance:
  - macro optimized: rating micro-F1 78.7%, macro-F1 77.5%
  - micro optimized: rating micro-F1 83.1%, macro-F1 81.8%
- training data: filtered Danbooru 2024 subset

評価観点:

- Danbooru 4分類 rating に近く、downstream mapper と相性がよい
- rating-only ではなく multi-label tagger なので重さ/速度を検証する
- 既存 tagger 経路として追加するか、rating extraction だけ使うかを判断する

### 4. `ggg4mless/RateBooru_Efficient`

- URL: https://huggingface.co/ggg4mless/RateBooru_Efficient
- labels: `general`, `questionable`, `explicit`
- base: TensorFlow/Keras EfficientNetB1
- input: 240x240 RGB
- dataset: 6,000 images, private/custom-collected Danbooru rating metadata
- validation metrics:
  - accuracy 70.42%
  - precision 72.93%
  - recall 67.58%

評価観点:

- label scheme は扱いやすい
- TensorFlow/Keras 依存なので Issue #73 の eager import 問題と相性が悪い
- dataset/精度の根拠が弱く、優先度は低め

### 5. 汎用 explicit-content classifier 系

例:

- `prithivMLmods/vit-mini-explicit-content`
- `CaveduckAI/nsfw-classifier`
- `Falconsai/nsfw_image_detection`
- `onnx-community/nsfw-image-detector-ONNX`

評価観点:

- 多くは moderation / binary NSFW classifier で、rating model ではない
- image-annotator-lib 側で追加するなら model-native label をそのまま返す
- LoRAIro 側では API 送信前の除外・ルーティング signal、または粗い既存 rating への割当として使う
- 新 DB カラム追加はしない方針

## 実装タスク

- [ ] 各候補のライセンス・依存・モデルサイズ・入力形式を確認する
- [ ] `deepghs/anime_rating` の最小 PoC を作る
- [ ] `Camais03/camie-tagger-v2` の速度/メモリ/出力 schema を確認する
- [ ] TensorFlow/Keras 系候補は Issue #73 との関係を確認し、低優先で評価する
- [ ] rating-only annotator を追加する場合も heavy dependency を import 時に eager load しない
- [ ] output shape は model-native: `raw_label`, `confidence_score`, `source_scheme`
- [ ] LoRAIro canonical rating への mapper は library へ入れない
- [ ] runtime validation test を追加する

## 受け入れ条件

- 新規候補を採用するか/見送るかの判断が記録されている
- 採用する場合、`annotate()` から model-native rating が取得できる
- LoRAIro 側で変換できる raw label / confidence / source scheme が返る
- import 時に重い ML framework を eager load しない
- 既存 WDTagger rating 昇格 (#80) と責務が重複しない

## 関連

- Parent: #79
- Existing implemented model support: #80
- LoRAIro-side mapping/save issue: NEXTAltair/LoRAIro#333

### PR DESCRIPTION
**Pull request by NEXTAltair**
## Summary
- add a rating-only ONNX annotator for deepghs/anime_rating mobilenetv3_sce_dist
- register anime_rating_mobilenetv3_sce_dist with ratings capability in runtime/sample/package configs
- document rating model candidate specs and add unit/runtime validation

## Tests
- uv run ruff check src/image_annotator_lib/model_class/rating_onnx.py src/image_annotator_lib/core/model_config.py tests/unit/model_class/test_rating_onnx.py tests/runtime_validation/test_real_model_runtime.py
- uv run ruff format --check src/image_annotator_lib/model_class/rating_onnx.py src/image_annotator_lib/core/model_config.py tests/unit/model_class/test_rating_onnx.py tests/runtime_validation/test_real_model_runtime.py
- uv run pytest tests/unit/model_class/test_rating_onnx.py tests/unit/test_capability_utils.py tests/unit/fast/test_list_annotator_info.py
- uv run pytest tests/runtime_validation/test_real_model_runtime.py::test_real_anime_rating_runtime -m downloads_and_runs_model

Closes #81

### CODE REVIEW

File: src/image_annotator_lib/model_class/rating_onnx.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        from huggingface_hub import hf_hub_download
+
+        model_file = hf_hub_download(model_path, f"{model_variant}/model.onnx")
+        meta_file = hf_hub_download(model_path, f"{model_variant}/meta.json")
+        return Path(model_file), Path(meta_file)
+
+    def _load_labels(self, meta_path: Path) -> list[str]:
+        try:
+            labels = json.loads(meta_path.read_text(encoding="utf-8")).get("labels")
+        except FileNotFoundError:
```
[the remark this answers is not in the document]
Comment (NEXTAltair) [reply src/image_annotator_lib/model_class/rating_onnx.py:87]: 方針として、今回の初期adapterでは公式配布の `deepghs/anime_rating` の `meta.json` をモデル配布物の一部として信頼します。

壊れた `meta.json` や途中で壊れたローカルcacheはあり得ますが、通常のHugging Face配布物利用ではかなり低頻度のローカル/offline運用問題だと見ています。まずはPRのscopeをrating annotator追加に絞り、実際の失敗報告が出た時点で defensive fallback を追加する方針にします。

File: src/image_annotator_lib/model_class/rating_onnx.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            logger.warning(f"Anime rating meta file not found: {meta_path}; fallback labels used")
+            return list(self.default_labels)
+
+        if not isinstance(labels, list) or not all(isinstance(label, str) for label in labels):
+            logger.warning(f"Invalid anime rating labels in {meta_path}; fallback labels used")
+            return list(self.default_labels)
```
[the remark this answers is not in the document]
Comment (NEXTAltair) [reply src/image_annotator_lib/model_class/rating_onnx.py:93]: ここも同じ方針です。

`labels` 数の不一致は、別variantの `meta.json` を手動で置いた、古いmetadataを指した、などのローカル配置ミスで起きるケースだと考えています。公式の `mobilenetv3_sce_dist/model.onnx` と `mobilenetv3_sce_dist/meta.json` の組み合わせでは問題ないことを確認済みなので、現時点では追加validationは入れず、実際に問題報告が出たら明確なエラー化またはfallbackを追加します。
